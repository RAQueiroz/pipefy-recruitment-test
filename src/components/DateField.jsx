import React from "react";
import styled from "styled-components";
import StyledLabel from "./StyledLabel";
import Colors from '../utils/Colors'
import {InlineError} from "./StyledElements";
const StyledInput = styled.input`
  border: solid 1px ${props => (props.error ? Colors.errorLight : "#ccc")};
  border-radius: 5px;
  line-height: 40px;
  padding-left: 10px;
`;


const DateField = ({ label, id, error, value = "", onChange = () => {} }) => (
  <div>
    <StyledLabel>
      {label}
      <StyledInput
        error={error}
        value={value}
        type="date"
        id={id}
        onChange={event => onChange({ id, value: event.target.value })}
      />

    </StyledLabel>
      {error && <InlineError>{error}</InlineError>}
  </div>
);

export default DateField;
