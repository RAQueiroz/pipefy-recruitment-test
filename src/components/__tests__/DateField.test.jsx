import React from "react";
import DateField from "../DateField";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(" <DateField/>", () => {
    const onChangeSpy = jest.fn();

    const item = {
        id: "xpto",
        label: "Name",
        value: "08/12/1986"
    };

    const wrapper = shallow(
        <DateField onChange={onChangeSpy} {...item} />
    );

    it("renders", () => {
        expect(wrapper.find(`#${item.id}`)).toHaveLength(1);
    });

    it("to match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("display label based on props", () => {
        expect(wrapper.containsMatchingElement(item.label)).toBeTruthy();
        wrapper.setProps({ label: "Birthday" });
        expect(wrapper.containsMatchingElement("Birthday")).toBeTruthy();
    });

    it("display value based on props", () => {
        expect(wrapper.find(`#${item.id}`).props().value).toEqual(
            item.value
        );
        wrapper.setProps({ value: "08/12/2017" });
        expect(wrapper.find(`#${item.id}`).props().value).toEqual("08/12/2017");
    });

    it("check if onChange works", () => {
        wrapper
            .find(`#${item.id}`)
            .simulate("change", { target: { value: '08/12/2018' } });
        expect(onChangeSpy).toBeCalledWith({ id: "xpto", value: "08/12/2018" });
    });
});
