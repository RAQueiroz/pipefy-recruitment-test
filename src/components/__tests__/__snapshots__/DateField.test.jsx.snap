// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <DateField/> to match snapshot 1`] = `
<div>
  <styled.label>
    Name
    <styled.input
      id="xpto"
      onChange={[Function]}
      type="date"
      value="08/12/1986"
    />
  </styled.label>
</div>
`;
